### DISCUSSION TITLE
`useFormStatus` not working when `onSubmit` is set

### DISCUSSION THREAD

**Discussion by ahkhanjani** [vercel/next.js#63228] [App Router]
When you set both `action` and `onSubmit`, `useFormStatus` won't work anymore. Is this intended functionality? I'm setting `onSubmit` to run a client-side validation with react-hook-form and zod, and call submit using form's ref.

**Comment by NomiAdam** [comment]
Same issue here, it's getting really unproductive working with newest Next and dealing with all these weird caveats
